main.cpp

```diff
@@ -73,29 +73,36 @@ struct <#structName1#>                                //4
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float name1 { 30 }, name2 { 20 };
+    float memberFunction(float* updatedValue)      //12
     {
-        
+        std::cout << "U's name1 value: " << name1 << std::endl;
+        name1 = *updatedValue;
+        std::cout << "U's name1 updated value: " << name1 << std::endl;
+        while( std::abs(name2 - name1) > 0.001f )
+        {
+            name1 -= .5f;
+            name2 -= 1.0f;
+        }
+        std::cout << "U's name2 updated value: " << name2 << std::endl;
+        return name2 * name1;        
     }
 };
 
-struct <#structname2#>
+struct struct2
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's name1 value: " << that->name1 << std::endl;
+        that->name1 = *updatedValue;
+        std::cout << "U's name1 updated value: " << that->name1 << std::endl;
+        while( std::abs(that->name2 - that->name1) > 0.001f )
```

Comment:
What do we always check before using pointers? 
- [ ] revise

Reply:
Always check to make sure your pointer is not NULL before using it!!!! :)